src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -1014,60 +773,47 @@ batt_smbus_main(int argc, char *argv[])
 	if (g_batt_smbus == nullptr) {
 		PX4_INFO("not started");
 		batt_smbus_usage();
-		return 1;
+		exit(1);
```

Comment:
Please return an error instead of exiting. On Linux these modules are threads within a single progress, so exiting from a single module/driver kills all of px4.

Reply:
Woops, I was working off of the older one and not your new commit today.